### PR TITLE
feat(campaign): reward dollars in token and captcha

### PR DESCRIPTION
**Pull request by nguyenhuudungz**
- campaign improvement, detail [here](https://www.notion.so/kybernetwork/Campaign-v2-Improvements-0a1c72109ca943db8de4a8b6a94f6eeb)

### CODE REVIEW

File: src/pages/Campaign/CampaignListAndSearch.tsx [hunk 1]
```diff
@@ -46,49 +47,55 @@ export default function CampaignListAndSearch({
...
           }, new Fraction(0))
 
+          const rCampaignName = campaign.name
+          const rCampaignStatus =
+            campaign.status === 'Upcoming' ? t`Upcoming` : campaign.status === 'Ongoing' ? t`Ongoing` : t`Ended`
+          const rChainIdImages =
+            campaign &&
+            campaign.chainIds &&
+            campaign.chainIds
+              .split(',')
```
Comment (namgold) [inline src/pages/Campaign/CampaignListAndSearch.tsx:68]: `campaign?.chainIds?.split?.(',')`

File: src/pages/Campaign/CampaignListAndSearch.tsx [hunk 2]
```diff
@@ -46,49 +47,55 @@ export default function CampaignListAndSearch({
...
+              .map(chainId => (
+                <img
+                  key={chainId}
+                  src={
+                    isDarkMode && !!NETWORKS_INFO[chainId as unknown as ChainId].iconDark
```
Comment (namgold) [inline src/pages/Campaign/CampaignListAndSearch.tsx:73]: should be `NETWORKS_INFO[chainId as unknown as ChainId]?.iconDark` to cover case non exists `chainId`

File: src/pages/Campaign/CampaignListAndSearch.tsx [hunk 3]
```diff
@@ -46,49 +47,55 @@ export default function CampaignListAndSearch({
...
+                      ? NETWORKS_INFO[chainId as unknown as ChainId].iconDark
+                      : NETWORKS_INFO[chainId as unknown as ChainId].icon
+                  }
+                  alt="network_icon"
```
Comment (namgold) [inline src/pages/Campaign/CampaignListAndSearch.tsx:77]: Should we provide more details? Like ``{NETWORKS_INFO[chainId as unknown as ChainId].name + ' icon'}``

File: src/pages/Campaign/CampaignListAndSearch.tsx [hunk 4]
```diff
@@ -46,49 +47,55 @@ export default function CampaignListAndSearch({
...
+                  style={{ width: '16px', minWidth: '16px', height: '16px', minHeight: '16px' }}
+                />
+              ))
+
+          const totalRewardAmountString = totalRewardAmount.toSignificant(DEFAULT_SIGNIFICANT, { groupSeparator: ',' })
+          const tokenSymbol = campaign.rewardDistribution[0]?.token?.symbol
+          const rCampaignReward = isRewardInUSD
+            ? t`$${totalRewardAmountString} ${tokenSymbol}`
+            : `${totalRewardAmountString} ${tokenSymbol}`
```
Comment (nguyenhoaidanh) [inline src/pages/Campaign/CampaignListAndSearch.tsx:86]:
```
t`$${totalRewardAmountString} ${tokenSymbol}`
vs
`${totalRewardAmountString} ${tokenSymbol}`
```
can we add `t` or remove `t`?

File: src/pages/Campaign/CampaignListAndSearch.tsx (hunk 2 above)
Comment (namgold) [reply src/pages/Campaign/CampaignListAndSearch.tsx:73]: nevermind, `NETWORKS_INFO` have a proxy to cover such case :joy:

File: src/pages/Campaign/CampaignListAndSearch.tsx (hunk 3 above)
Comment (nguyenhuudungz) [reply src/pages/Campaign/CampaignListAndSearch.tsx]: done

File: src/components/ProgressBar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+}: {
+  title: string
+  value: string
+  percent: number
+  color?: string
+  valueTextColor?: string
+}) {
+  const theme = useTheme()
+  return (
+    <Flex flexDirection={'column'} style={{ gap: 5 }}>
```
Comment (nguyenhuudungz) [inline src/components/ProgressBar/index.tsx:38]: I usually use the double quote here.

File: src/pages/Campaign/CampaignButtonEnterNow.tsx [hunk 6]
```diff
@@ -40,39 +36,28 @@ export default function CampaignButtonEnterNow() {
     try {
       setLoading(true)
       const token = await executeRecaptcha('enterCampaign')
+      console.log(`token`, token)
```
Comment (nguyenhuudungz) [inline src/pages/Campaign/CampaignButtonEnterNow.tsx:39]: Can you remove this for me?

**Review by namgold — APPROVED** [review]
lgtm

File: src/components/ProgressBar/index.tsx (hunk 5 above)
Comment (nguyenhoaidanh) [reply src/components/ProgressBar/index.tsx:38]: it's autocompleted bracket for me.

**Review by namgold — APPROVED** [review]

**Review by viet-nv — APPROVED** [review]

**Review by nguyenhoaidanh — APPROVED** [review]

**Pull request merged**